From 72db525aaccc1a2f293d3caaa1cc7b59e83a788b Mon Sep 17 00:00:00 2001
From: SamHurwitz <samhurwitz@mail.ru>
Date: Sat, 17 Oct 2020 14:58:31 +0500
Subject: [PATCH 03/85] Fix manager not passing -i option

---
 src/manager.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/manager.c b/src/manager.c
index fd3c645..c38d252 100644
--- a/src/manager.c
+++ b/src/manager.c
@@ -186,6 +186,10 @@ construct_command_line(struct manager_ctx *manager, struct server *server)
         int len = strlen(cmd);
         snprintf(cmd + len, BUF_SIZE - len, " -u");
     }
+    if (manager->iface) {
+        int len = strlen(cmd);
+        snprintf(cmd + len, BUF_SIZE - len, " -i \"%s\"", manager->iface);
+    }
     if (server->fast_open[0] == 0 && manager->fast_open) {
         int len = strlen(cmd);
         snprintf(cmd + len, BUF_SIZE - len, " --fast-open");
-- 
2.25.1

