From 920dbac0d38d779cb6d4aed0857990d8ae9709af Mon Sep 17 00:00:00 2001
From: Not Sure <notsure2@protonmail.com>
Date: Tue, 22 Dec 2020 09:14:04 +0200
Subject: [PATCH 06/85] Missing config file parsing logic.

---
 src/jconf.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/jconf.c b/src/jconf.c
index 5adba34..765a3e2 100644
--- a/src/jconf.c
+++ b/src/jconf.c
@@ -274,6 +274,14 @@ read_jconf(const char *file)
                 check_json_value_type(value, json_boolean,
                                       "invalid config file: option 'reuse_port' must be a boolean");
                 conf.reuse_port = value->u.boolean;
+            } else if (strcmp(name, "tcp_incoming_sndbuf") == 0) {
+                check_json_value_type(value, json_integer,
+                                      "invalid config file: option 'tcp_incoming_sndbuf' must be an integer");
+                conf.tcp_incoming_sndbuf = value->u.integer;
+            } else if (strcmp(name, "tcp_outgoing_sndbuf") == 0) {
+                check_json_value_type(value, json_integer,
+                                      "invalid config file: option 'tcp_outgoing_sndbuf' must be an integer");
+                conf.tcp_outgoing_sndbuf = value->u.integer;
             } else if (strcmp(name, "auth") == 0) {
                 FATAL("One time auth has been deprecated. Try AEAD ciphers instead.");
             } else if (strcmp(name, "nofile") == 0) {
-- 
2.25.1

