From 8bcffa54b58411a2beda6acabd5b849518c66841 Mon Sep 17 00:00:00 2001
From: Gilles Depeyrot <g.depeyrot@transacom.fr>
Date: Thu, 15 Jul 2021 14:43:22 +0200
Subject: [PATCH 24/85] fix check for return value of open when redirecting to
 null device

---
 src/utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils.c b/src/utils.c
index 4abeb59..c8daebb 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -484,7 +484,7 @@ daemonize(const char *path)
     }
 
     int dev_null = open("/dev/null", O_WRONLY);
-    if (dev_null) {
+    if (dev_null > 0) {
         /* Redirect to null device  */
         dup2(dev_null, STDOUT_FILENO);
         dup2(dev_null, STDERR_FILENO);
-- 
2.25.1

