From b9117559000e1bdee4a84383c537ca5b4b7fe8dc Mon Sep 17 00:00:00 2001
From: Max Lv <mlv@nvidia.com>
Date: Thu, 19 Aug 2021 08:38:02 +0800
Subject: [PATCH 29/85] Remove the fragmentation check for the first AEAD chunk

---
 src/server.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/server.c b/src/server.c
index 6447e84..e9cdc26 100644
--- a/src/server.c
+++ b/src/server.c
@@ -85,10 +85,6 @@ enum datatypes {
 #define SSMAXCONN 1024
 #endif
 
-#ifndef MAX_FRAG
-#define MAX_FRAG 1
-#endif
-
 #ifdef USE_NFCONNTRACK_TOS
 
 #ifndef MARK_MAX_PACKET
@@ -982,11 +978,6 @@ server_recv_cb(EV_P_ ev_io *w, int revents)
         return;
     } else if (err == CRYPTO_NEED_MORE) {
         if (server->stage != STAGE_STREAM) {
-            if (server->frag > MAX_FRAG) {
-                report_addr(server->fd, "malicious fragmentation");
-                stop_server(EV_A_ server);
-                return;
-            }
             server->frag++;
         }
         return;
-- 
2.25.1

