From c2fc967c1a3e4d864d866cc92dd29a04cdb11f98 Mon Sep 17 00:00:00 2001
From: Bill Zhong <billzhong@users.noreply.github.com>
Date: Fri, 27 May 2022 11:06:37 -0230
Subject: [PATCH 36/85] fix .copr build

---
 .copr/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/.copr/Makefile b/.copr/Makefile
index f88794f..7a74155 100644
--- a/.copr/Makefile
+++ b/.copr/Makefile
@@ -16,4 +16,5 @@ deps:
 endif
 
 srpm: deps
-	$(TOP_DIR)/rpm/genrpm.sh -o $(outdir)
+	git config --global --add safe.directory $(TOP_DIR)
+	$(RPM_DIR)/genrpm.sh -o $(outdir)
-- 
2.25.1

