From 549b9bedb9fa3cc2c98febd0a8cff9c307dd3559 Mon Sep 17 00:00:00 2001
From: Maxim Galaganov <max@internet.ru>
Date: Fri, 24 Jun 2022 13:21:01 +0300
Subject: [PATCH 40/85] Fix rpm build for fedora 36

---
 rpm/SPECS/shadowsocks-libev.spec.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rpm/SPECS/shadowsocks-libev.spec.in b/rpm/SPECS/shadowsocks-libev.spec.in
index 5be416d..66949b7 100644
--- a/rpm/SPECS/shadowsocks-libev.spec.in
+++ b/rpm/SPECS/shadowsocks-libev.spec.in
@@ -14,6 +14,9 @@
 %global use_systemd 0
 %endif
 
+# https://fedoraproject.org/wiki/Changes/RemoveLaFiles
+%global __brp_remove_la_files %nil
+
 Name:		    @NAME@
 Version:	    @VERSION@
 Release:	    @RELEASE@%{?dist}
-- 
2.25.1

