From 05e70d43176ae239ba54ffb1a0f80df5b8f3d4f1 Mon Sep 17 00:00:00 2001
From: DDoSolitary <DDoSolitary@gmail.com>
Date: Sun, 13 Mar 2022 15:59:18 +0800
Subject: [PATCH 41/85] fix c-ares link dep

---
 configure.ac       | 2 +-
 src/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 84d6e82..134ea5f 100755
--- a/configure.ac
+++ b/configure.ac
@@ -66,7 +66,7 @@ dnl Add library for mingw
 case $host in
   *-mingw*)
     CFLAGS="$CFLAGS -mno-ms-bitfields"
-    LIBS="$LIBS -lws2_32"
+    LIBS="$LIBS -lws2_32 -liphlpapi"
     ;;
   *-cygwin*)
     CFLAGS="$CFLAGS -mno-ms-bitfields"
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 346ba07..9ca9d85 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -117,7 +117,7 @@ list(APPEND DEPS
         )
 
 if (MINGW)
-list(APPEND DEPS ws2_32)
+list(APPEND DEPS ws2_32 iphlpapi)
 add_compile_definitions(CARES_STATICLIB PCRE_STATIC)
 endif ()
 endif ()
-- 
2.25.1

